### PR TITLE
fix Residual Filter containing wrong info

### PR DESCRIPTION
**Pull request by zhexuany**
### What problem does this PR solve? 
This PR fixes explain info having wrong information.

### CODE REVIEW

**Comment by marsishandsome** [comment]
need cherry-pick ?

**Comment by birdstorm** [comment]
add some tests?

**Comment by marsishandsome** [comment]
two `fix` on title ?

**Comment by birdstorm** [comment]
consider move the test into PlanTestSuite?

File: core/src/main/scala/com/pingcap/tispark/utils/TiUtil.scala [hunk 1]
```diff
@@ -243,4 +244,20 @@ object TiUtil {
 
     mutableRow
   }
+
+  def extractDAGReq(df: DataFrame): TiDAGRequest = {
+    val executedPlan = df.queryExecution.executedPlan
+    val copRDD = executedPlan.find(e => e.isInstanceOf[ColumnarCoprocessorRDD])
+    val regionTaskExec = executedPlan.find(e => e.isInstanceOf[ColumnarCoprocessorRDD])
```
Comment (marsishandsome) [inline core/src/main/scala/com/pingcap/tispark/utils/TiUtil.scala]: `copRDD` is same as `regionTaskExec` ?

File: core/src/main/scala/com/pingcap/tispark/utils/TiUtil.scala [hunk 2]
```diff
@@ -243,4 +244,20 @@ object TiUtil {
 
     mutableRow
   }
+
+  def extractDAGReq(df: DataFrame): TiDAGRequest = {
```
Comment (marsishandsome) [inline core/src/main/scala/com/pingcap/tispark/utils/TiUtil.scala]: what if there are more than one or no `TiDAGRequest` in `df` ? show we return `List<TiDAGRequest>` ?

File: core/src/main/scala/com/pingcap/tispark/utils/TiUtil.scala (hunk 1 above)
Comment (marsishandsome) [reply core/src/main/scala/com/pingcap/tispark/utils/TiUtil.scala]: change to `val regionTaskExec = executedPlan.find(e => e.isInstanceOf[ColumnarRegionTaskExec])` ?

File: core/src/main/scala/com/pingcap/tispark/utils/TiUtil.scala (hunk 2 above)
Comment (zhexuany) [reply core/src/main/scala/com/pingcap/tispark/utils/TiUtil.scala]: We should only have one TiDAGRequest.

File: core/src/test/scala/org/apache/spark/sql/IssueTestSuite.scala [hunk 3]
```diff
@@ -16,6 +16,7 @@
 package org.apache.spark.sql
 
 import com.pingcap.tispark.TiConfigConst
+import com.pingcap.tispark.utils.TiUtil
```
Comment (marsishandsome) [inline core/src/test/scala/org/apache/spark/sql/IssueTestSuite.scala]: unused import?

File: core/src/main/scala/com/pingcap/tispark/utils/TiUtil.scala (hunk 2 above)
Comment (marsishandsome) [reply core/src/main/scala/com/pingcap/tispark/utils/TiUtil.scala]: maybe zero ?

Comment (zhexuany) [reply core/src/main/scala/com/pingcap/tispark/utils/TiUtil.scala]: Originally, it is zero while IntelliJ gives me a warning.

**Pull request merged**